web-console/4.0-designs/code-editors/code-editor.md

```diff
@@ -0,0 +1,46 @@
+# Code Editor
+
+## Edit YAML
+
+![pic 1](img/1.png)
+
+- The code area is borderless at the top and has the button bar at the bottom as a separator.
+- The Edit action can be removed from the Actions dropdown on every detail page. Users can access the tab as needed.
+- The Download and Expand actions are displayed as links in the top right corner above the YAML editor, with a vertical separator between them. The icons belong to the left of each link.
```

Comment:
In current design/implementation, there are some objects we provide samples for handholding users to create/edit them, e.g. `Network Policies`, `Roles`, and `BuildConfigs`. In both Create/Edit YAML editor, there's a sidebar (default present/expanded) on the right and shows a list of examples. 

*Create BuildConfig - sidebar expanded (default):*
<img width="1404" alt="create-buildconfig___sidebar-expanded" src="https://user-images.githubusercontent.com/5903705/43364955-74dc96f8-92d9-11e8-95fd-2caa31c07e61.png">

*Create BuildConfig - sidebar collapsed:*
<img width="1404" alt="create-buildconfig___sidebar-collapsed" src="https://user-images.githubusercontent.com/5903705/43364959-7c977124-92d9-11e8-96ca-e366c03a92f9.png">

*Edit BuildConfig - sidebar expanded (default):*
<img width="1403" alt="edit-buildconfig___sidebar-expanded" src="https://user-images.githubusercontent.com/5903705/43364962-854e9cf2-92d9-11e8-823b-3fe80c3657f1.png">

*Edit BuildConfig - sidebar collapsed:*
<img width="1403" alt="edit-buildconfig___sidebar-collapsed" src="https://user-images.githubusercontent.com/5903705/43364969-8c71aff6-92d9-11e8-8b1d-6410ee82c2b2.png">


When sidebar is collapsed (click on the "close" icon), the "View samples" link is designed as `position: absolute` (sticky position). By this way, it floats on the screen when scrolling, so we can safe a dedicated row here.

Since the "Download | Expand" actions button links are quite similar to this "View samples" link. I wonder if we can reuse the similar pattern here? I can come up some quick mocks for discussions.